### PR TITLE
Fix troubles to push ssl custom cert on android 10 unrooted device

### PR DESCRIPTION
**Pull request by cmajed**
Update dockerfile by removing all unnecessary stuff

Update work script: adding adb rooting command and making it simpler

Update README file according to previous changes

### CODE REVIEW

File: package.sh [hunk 1]
```diff
@@ -1,20 +1,11 @@
...
-    fullchain=$2
-    ssh -oServerAliveInterval=1 -oServerAliveCountMax=2 -oStrictHostKeyChecking=no -N -i genymotion_aws.pem -L 5555:localhost:5555 root@${domain} &
 
-    # Wait for ssh ...
-    sleep 5
-else
-    privkey=/etc/letsencrypt/live/${domain}/privkey.pem
-    fullchain=/etc/letsencrypt/live/${domain}/fullchain.pem 
+privkey=$1
+fullchain=$2
```
Comment (gvercoutere-geny) [inline package.sh:18]: ajouter une vérification si privkey et fullchain ne sont pas des fichiers?

File: package.sh [hunk 2]
```diff
@@ -27,9 +18,14 @@ rm tmp.pkcs12
...
 # Hardwork is done, give useful stuff to Android
+adb connect ${domain}:5555
+sleep 1
+adb root
 adb shell setprop persist.tls.pw.ks ${keystorepassword}
 adb shell setprop persist.tls.pw.cert ${certpassword}
-adb shell mkdir /data/misc/tls/
+adb shell mkdir -p /data/misc/tls/
 adb push custom.bks /data/misc/tls/custom.bks
 adb shell sync
```
Comment (gvercoutere-geny) [inline package.sh:49]: maybe push/sync before setprop, if mkdir or push fail

**Comment by gvercoutere-geny** [comment]
LGTM. Maybe add some checks when it is possible.

**Review by gvercoutere-geny — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -26,11 +26,9 @@ Adb should be enabled on your instance.
 
 * Run this docker
 
-`docker run --net=host -P -v <path_to_genymotion_cloud_ssl_tool>:/opt/work/ --env package=true --env keystorepassword=<kspassword> --env certpassword=<kppassword> -t -i <name> /opt/work/package.sh <relpath_to_privkey.pem> <relpath_to_fullchain.pem>`
+`docker run -v <full_path_to_genymotion_cloud_ssl_tool>:/opt/work/ --env package=true --env keystorepassword=<kspassword> --env certpassword=<kppassword> --env domain=<domainName> -t -i <name>
```
Comment (c-quenin) [inline README.md]: ` is missing at the end of the docker run command

Also, detail, but I find the synthax with ``` blocks more readable for README commands

File: Dockerfile [hunk 4]
```diff
@@ -1,37 +1,23 @@
-# SSL Certificate generation with let's encrypt
-
 # Before running this, your <domain> DNS should already point to your
-# AWS instance IP and your AWS security group should allow let's encrypt servers
+# AWS instance IP and you already generated a valid certificate for your DNS
```
Comment (c-quenin) [inline Dockerfile]: I think this kind of info should really appear in the README instead, I think most users won't even bother opening the Dockerfile and these steps are mandatory.

I'd remove all these comments and edit the README instead

File: package.sh [hunk 5]
```diff
@@ -27,9 +41,14 @@ rm tmp.pkcs12
...
+adb connect ${domain}:5555
+sleep 1
+adb root
+adb shell mkdir -p /data/misc/tls/
 adb push custom.bks /data/misc/tls/custom.bks
 adb shell sync
+adb shell setprop persist.tls.pw.ks ${keystorepassword}
+adb shell setprop persist.tls.pw.cert ${certpassword}
 
+# A last Clean up
```
Comment (c-quenin) [inline package.sh]: mini typo: Clean -> clean

File: Dockerfile [hunk 6]
```diff
@@ -1,37 +1,23 @@
...
 # build this docker:
-    # docker build -t <usr>/<name> .
-
-# To use existing certificate
-# docker run --net=host -P -v <path_to_aws_ssl>:/opt/work/ --env domain=<domain> --env package=true --env keystorepassword=<kspassword> --env certpassword=<kppassword> -t -i <usr>/<name> /opt/work/package.sh <relpath_to_privkey.pem> <relpath_to_fullchain.pem>
+    # docker build -t <name> .
 
-# Alternatively, you could install those dependencies on your own distro, set environnment variables
-# according to the 'docker run' line and run generate.sh
+# copy your cert and key files to genymotion-cloud-ssl-tool folder
```
Comment (c-quenin) [inline Dockerfile]: Missing a little capital letter to copy -> Copy

**Review by c-quenin — COMMENTED** [review]
Minor comments, very nice cleanup :+1:

File: README.md [hunk 7]
```diff
@@ -2,35 +2,35 @@
 
 ## Prerequisites
 
-Adb should be enabled on your instance.
 
-## In case you already have your certificate
+1. Your `domain` DNS should already point to your AWS instance IP.
+2. You already generated a valid certificate for your `domain`.
+3. Adb should be enabled on your instance.
```
Comment (c-quenin) [inline README.md:8]: Also add '4. Your device must be rooted.'
Maybe also add the link to https://docs.genymotion.com/paas/10_Using_root_access/

File: README.md [hunk 8]
```diff
@@ -2,35 +2,35 @@
...
 
+## HOWTO
 * Clone this repository
 
-`git clone https://github.com/Genymobile/genymotion-cloud-ssl-tool.git`
+```git clone https://github.com/Genymobile/genymotion-cloud-ssl-tool.git```
```
Comment (c-quenin) [inline README.md]: For ``` blocks to appear, you need to jump a line, see [here](https://github.com/Genymobile/genymotion-cloud-ssl-tool/tree/SYSTEM-1981/fix_push_custom_ssl_cert).

File: README.md [hunk 9]
```diff
@@ -2,35 +2,35 @@
...
 
 * Go to `genymotion-cloud-ssl-tool` repository
 
-`cd genymotion-cloud-ssl-tool`
+```cd genymotion-cloud-ssl-tool```
 
 * Copy your privkey.pem and fullchain.pem to the `genymotion-cloud-ssl-tool` repository
 
-`cp /path/to/privkey.pem . && cp /path/to/fullchain.pem .`
+```cp /path/to/privkey.pem . && cp /path/to/fullchain.pem ```
```
Comment (c-quenin) [inline README.md]: a dot has disappeared at the end of the second command

File: README.md [hunk 10]
```diff
@@ -2,35 +2,35 @@
...
 
 * Build this docker
 
-`docker build -t <name> .`
+```docker build -t <name> ```
```
Comment (c-quenin) [inline README.md]: same thing here, a dot is missing at the end

**Review by guillaume-roche — APPROVED** [review]

**Review by c-quenin — APPROVED** [review]

**Pull request merged**